src/Microdown/MicHTTPResource.class.st

```diff
@@ -7,24 +7,25 @@ Class {
 	#name : #MicHTTPResource,
 	#superclass : #MicAbsoluteResource,
 	#classVars : [
-		'ImageCache'
+		'ImageCache',
+		'ImageNotCachedBlock'
 	],
 	#category : #'Microdown-Core'
 }
 
 { #category : #images }
 MicHTTPResource class >> cachedImageFromUrl: aUrl [
+	"In case a url is not in the cache, it is either read from the net, or if the ImageNotCachedBlock
+	is non nil, the image returned by ImageNotCachedBlock is returned, but not stored in the cache"
 	
-	^ self imageCache 
-		at: aUrl
-		ifAbsentPut: [ (self rawImageFrom: aUrl) ifNil: [ ^ nil ] ] 
-]
-
-{ #category : #accessing }
-MicHTTPResource class >> imageAt: urlString put: aForm [	
-	"the cache contains results of for example ZnEasy getPng:, i.e., a form"
-	
-	ImageCache at: urlString put: aForm
+	(self imageCache includesKey: aUrl)
+		ifTrue: [ ^ self imageCache at: aUrl ].
+		
+	ImageNotCachedBlock 
```

Comment:
I found the logic with nil instead of boolean for ImageNotCachedBlock difficult to follow. 
I could not tell you if ifNil is caching or not. 
BTW you do not need the return in the ifNil: and ifNotNil: block

Reply:
I tend to agree that is a bit strange - I tried to tell what it was by adding "Block" at the end. It is a way in which HTTP resource avoids supporting Beautiful comments directly, but to give BC the functionality it need to do its offline stuff. The actual block is set in BC as part of its Offline handling, and is nil by default.

Reply:
I will look at the ifNil thingie